processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -79,6 +81,11 @@ func NewDetector(set processor.Settings, dcfg internal.DetectorConfig) (internal
 
 func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schemaURL string, err error) {
 	if _, err = d.metadataProvider.InstanceID(ctx); err != nil {
+		var requestSendError *smithyhttp.RequestSendError
+		if errors.As(err, &requestSendError) {
+			d.logger.Warn("EC2 metadata endpoint unreachable", zap.Error(err))
+			return pcommon.NewResource(), "", err
```

Comment:
Can you please add a comment here saying why we return an error for this particular use case?

Also, are we sure that `smithyhttp.RequestSendError` is fired only when ec2 is in _not ready_ state? Will it be a different error if it's not an ec2 instance?

Also, `smithyhttp.RequestSendError` might be unreliable. If the SDK is changed to use something else, we might miss it.

I'm thinking if we can have a configuration option instead. Happy to here other opinions.

Reply:
This looks good to me as the initial solution, if the first 2 Qs above are addressed

Reply:
I think I can work on adding an integration test to make sure this works as intended.

Reply:
> Can you please add a comment here saying why we return an error for this particular use case?

I put a log warning above, can I maybe make that warning better, maybe more actionable? I will do that first. Comments tend to rot...

Reply:
> I put a log warning above, can I maybe make that warning better, maybe more actionable? I will do that first. Comments tend to rot...

The warning is for the end user. When someone reads the code, it's unclear _why_ this particular error should be propagated while others are not.